drivers/rtc/rtc_max31329.c

```diff
@@ -0,0 +1,570 @@
+	} else {
+		data->alarm_callback(data->dev, 0, data->alarm_user_data);
+	}
+}
+
+/* The function called when the clock alarm activates the interrupt*/
+void gpio_callback_function(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	struct max31329_data *data = CONTAINER_OF(cb, struct max31329_data, intb_callback);
+
+	LOG_DBG("MAX31329 interrupt detected");
+	/* By using a work we are able to run "heavier" code */
+	k_work_submit(&(data->callback_work));
+}
+#endif
+
+static int max31329_alarm_set_callback(const struct device *dev, uint16_t id,
+				       rtc_alarm_callback callback, void *user_data)
+{
+#ifndef MAX31329_INT_GPIOS_IN_USE
+	ARG_UNUSED(dev);
+	ARG_UNUSED(id);
+	ARG_UNUSED(callback);
+	ARG_UNUSED(user_data);
+
+	return -ENOTSUP;
+#else
+	const struct max31329_config *config = dev->config;
+	struct max31329_data *data = dev->data;
+	int ret;
+
+	if (config->int_rtc.port == NULL) {
+		return -ENOTSUP;
+	}
+
+	if (id != 0) {
+		LOG_ERR("invalid ID %d", id);
+		return -EINVAL;
+	}
+
+	data->alarm_callback = callback;
+	data->alarm_user_data = user_data;
+	data->dev = dev;
+
+	ret = gpio_pin_configure_dt(&config->int_rtc, GPIO_INPUT);
+	if (ret < 0) {
+		LOG_ERR("Error %d: failed to configure %s pin %d", ret, config->int_rtc.port->name,
+			config->int_rtc.pin);
+		return ret;
+	}
+
+	ret = gpio_pin_interrupt_configure_dt(&config->int_rtc, GPIO_INT_EDGE_FALLING);
+	if (ret < 0) {
+		LOG_ERR("Error %d: failed to configure interrupt on %s pin %d", ret,
+			config->int_rtc.port->name, config->int_rtc.pin);
+		return ret;
+	}
+
+	gpio_init_callback(&data->intb_callback, gpio_callback_function, BIT(config->int_rtc.pin));
+	gpio_add_callback(config->int_rtc.port, &data->intb_callback);
+	LOG_DBG("Alarm set");
+
+	return 0;
+#endif
+}
+
+static const struct rtc_driver_api max31329_driver_api = {
+	.set_time = max31329_set_time,
+	.get_time = max31329_get_time,
+#ifdef CONFIG_RTC_ALARM
+	.alarm_get_supported_fields = max31329_alarm_get_supported_fields,
+	.alarm_set_time = max31329_alarm_set_time,
+	.alarm_get_time = max31329_alarm_get_time,
+	.alarm_is_pending = max31329_alarm_is_pending,
+	.alarm_set_callback = max31329_alarm_set_callback,
+#endif
+};
+
+int max31329_init(const struct device *dev)
+{
+	const struct max31329_config *config = dev->config;
+	int ret;
+	uint8_t reg_value;
+
+	/* Check if the I2C bus is ready*/
+	if (!device_is_ready(config->i2c.bus)) {
+		LOG_ERR("Failed to get pointer to %s device!", config->i2c.bus->name);
+		return -EINVAL;
+	}
+
+	/* Check if the device is alive by reading the status register*/
+	ret = i2c_reg_read_byte_dt(&config->i2c, MAX31329_REG_STATUS, &reg_value);
+	if (ret) {
+		LOG_ERR("Failed to read from MAX31329! (err %i)", ret);
+		return -EIO;
+	}
+
+	/* Disable trickle charging and power fail by default*/
+	ret = i2c_reg_write_byte_dt(&config->i2c, MAX31329_REG_PWR_MGMT, 0x00);
+	if (ret) {
+		LOG_ERR("Failed to disable power fail! (err %i)", ret);
+		return ret;
+	}
```

Comment:
Why the choice of disabling, then potentially re-enabling features, would just setting the value directly once not be simpler and cleaner?
```
        /* configure trickle charge */
	reg_value = config->trickle_charge_spec;
	reg_value |= config->trickle_charge_spec ? BIT(7) : 0;
	ret = i2c_reg_write_byte_dt(&config->i2c, MAX31329_REG_TRICKLE_REG, reg_value);
	if (ret) {
		LOG_ERR("Failed to configure trickle charging! (err %i)", ret);
		return ret;
	}
```
this costs only one I2C transaction :)